File: frontend/src/types/board.ts
export interface Board {
  id: number;
  title: string;
  content: string;
  writer: string;
  viewCount: number;
  createdAt: string;
  updatedAt: string;
}

export interface PageResponse<T> {
  content: T[];
  page: number;
  size: number;
  totalCount: number;
  totalPages: number;
}

export interface BoardCreateRequest {
  title: string;
  content: string;
  writer: string;
}

export interface BoardUpdateRequest {
  title: string;
  content: string;
}

File: backend/src/main/java/com/board/backend/board/domain/Board.java
package com.board.backend.board.domain;

import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;

@Getter
@Setter
public class Board {
    private Long id;
    private Long memberId;
    private String title;
    private String content;
    private String writer;
    private Integer viewCount;
    private Integer commentCount;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}
